app/models/user.rb

```diff
@@ -199,6 +201,28 @@ def username_required?
   def email_required?
     !erased?
   end
+  
+  def has_official_email?
+    domain = Setting.value_for 'email_domain_for_officials'
+    return false if !email or !domain or domain.length == 0
+    (email.end_with? "@#{domain}") or (email.end_with? ".#{domain}")
+  end
+  
+  # Check if the user is confirmed and has an official email address
+  # In that case, we assign a level 1 official level
+  def check_if_official_email
```

Comment:
Comments are good used sparsely for very complicated code.
In this case you could split the method in two and convey the same meaning.
For example: set_official_position and official_email?

(Instead of using if in a method it's usually nicer to use a ? sign a the end of the method.)
These two methods: has_official_email? and check_if_official_email imply the same meaning. They could probably be merged into a single method or split into more differentiated methods.